test/integration/mqtt_paho_binding/mqtt_test.go

```diff
@@ -0,0 +1,140 @@
+package mqtt_paho_binding
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/eclipse/paho.golang/paho"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cloudevents/sdk-go/protocol/mqtt_paho/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+
+	. "github.com/cloudevents/sdk-go/v2/binding/test"
+	. "github.com/cloudevents/sdk-go/v2/test"
+)
+
+const (
+	Broker = "127.0.0.1:1883"
+)
+
+type receiveMsg struct {
+	msg binding.Message
+	err error
+}
+
+func TestSendStructuredMessageToStructured(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	s, r, c, err := getSenderReceiver(ctx, cancel)
+	require.NoError(t, err)
+	defer c()
+
+	EachEvent(t, Events(), func(t *testing.T, eventIn event.Event) {
+		eventIn = ConvertEventExtensionsToString(t, eventIn)
+		in := MustCreateMockStructuredMessage(t, eventIn)
+
+		sendReceive(t, binding.WithPreferredEventEncoding(ctx, binding.EncodingStructured), s, r, in,
```

Comment:
do we really need that many levels of indirection? can't we keep the whole code within the test function? I have a feeling we're overcomplicating the test

Reply:
I try to reuse `sendReceive` in `TestSendStructuredMessageToStructured` and `TestSendBinaryMessageToBinary`. 

Of course, we can expand these codes and put them here. In this case, there will be some redundant codes in these two tests.
If you prefer to expand it, please reply directly to me and I can remove this layer from here.